### ISSUE TITLE
Dedicated code embed / preview component

### ISSUE THREAD

**Issue by fennifith** [unicorn-utterances/unicorn-utterances#388]
Many of our posts use stackblitz, codepen, or other external code embeds to show interactive previews of code snippets or allow readers to run the code and see a result. However, this can significantly slow down the loading speeds of these pages, and lacks some customizability in terms of styling or appearance.

I've additionally noticed that many uses of code blocks have a need to communicate a filename or directory structure in addition to their content, for which there is not currently any provided styling.

A specific code embed + preview component would allow for these snippets to be written inline with the post markdown, include a static result preview, and optionally link to an interactive codepen/stackblitz page where the user can edit or experiment with it further.

Proposed syntax / rough mockup for this component:

````markdown
<!-- code:start -->
# src/index.js [Edit on CodePen](https://codepen.io/example)

```js
console.log("Hello!");
```

# Result

Hello!
<!-- code:end -->
````

![2022-07-10-143527_768x204_scrot](https://user-images.githubusercontent.com/13000407/178157653-bf845bb9-9835-4582-9661-4ab193f95e2a.png)

Because this uses static content, it could also allow for other uses of this component than what codepen or stackblitz can enable (e.g. in the Minecraft datapacks article, different "results" like tables or images showing the results of different commands could be formatted here).

**Comment by fennifith** [comment]
I agree, the headings could be entirely removed. In my syntax example, I was more thinking about simplicity of implementation (and allowing the writer more control over the formatting) with the way it's laid out - the contents could effectively be split around the first code block, so everything before it is in the "header" and everything after in the "results".

e.g. that would also make the following possible:
````markdown
<!-- code:start -->
[src/index.js](https://github.com/...)
*Note: This snippet only contains part of the full code*

```js
doSomething();
```

<details>
<summary>Hidden Result</summary>

**CSV Output:**
| Lorem | Ipsum |
| ----- | ----- |
| Lorem | Ipsum |
| Lorem | Ipsum |

</details>
<!-- code:end -->
````

**Comment by crutchcorn** [comment]
We have the mockups for this from @PrattiDev for a v2 of the site:

<img width="368" alt="image" src="https://github.com/unicorn-utterances/unicorn-utterances/assets/9100169/34955ff6-0322-42d4-b890-6ba57a8d4c4c">

And can probably combine them without a browser preview like so:

<img width="466" alt="image" src="https://github.com/unicorn-utterances/unicorn-utterances/assets/9100169/fa54e10c-2f2e-4111-947c-e62727494ce2">
